[@reduxjs/toolkit: 1.6.2 → 1.7.0] TS2344: Type 'Definitions[EndpointName]' does not satisfy the constraint 'QueryDefinition<any, any, any, any, string>'.

I ran the `tsc` command immediately after updating reduxjs/toolkit from 1.6.2 to 1.7.0, and encountered this error.  

There is no way I can read this and understand it, can anyone tell me the cause or give me a clue?

```csharp
Error: node_modules/@reduxjs/toolkit/dist/query/core/module.d.ts(100,183): error TS2344: Type 'Definitions[EndpointName]' does not satisfy the constraint 'MutationDefinition<any, any, any, any, string>'.
  Type 'Definitions[MutationKeys<Definitions>]' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
    Type 'Definitions[Definitions[keyof Definitions] extends MutationDefinition<any, any, any, any, string> ? keyof Definitions : never]' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
      Type 'Definitions[keyof Definitions]' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
        Type 'Definitions[string] | Definitions[number] | Definitions[symbol]' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
          Type 'Definitions[string]' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
            Type 'EndpointDefinition<any, any, any, any, string>' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
              Type 'EndpointDefinitionWithQueryFn<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & QueryExtraOptions<any, any, any, any, string>' is not assignable to type 'MutationDefinition<any, any, any, any, string>'.
                Type 'EndpointDefinitionWithQueryFn<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & QueryExtraOptions<any, any, any, any, string>' is not assignable to type 'EndpointDefinitionWithQueryFn<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                  Type 'EndpointDefinitionWithQueryFn<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & QueryExtraOptions<any, any, any, any, string>' is not assignable to type 'MutationExtraOptions<any, any, any, any, string>'.
                    Types of property 'type' are incompatible.
                      Type 'DefinitionType.query' is not assignable to type 'DefinitionType.mutation'.
                        Type 'Definitions[string]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                          Type 'Definitions[keyof Definitions]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                            Type 'Definitions[Definitions[keyof Definitions] extends MutationDefinition<any, any, any, any, string> ? keyof Definitions : never]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                              Type 'Definitions[MutationKeys<Definitions>]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                Type 'Definitions[EndpointName]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                  Type 'Definitions[MutationKeys<Definitions>]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                    Type 'Definitions[Definitions[keyof Definitions] extends MutationDefinition<any, any, any, any, string> ? keyof Definitions : never]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                      Type 'Definitions[keyof Definitions]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                        Type 'Definitions[string] | Definitions[number] | Definitions[symbol]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                          Type 'Definitions[string]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                            Type 'EndpointDefinition<any, any, any, any, string>' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                              Type 'EndpointDefinitionWithQueryFn<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & QueryExtraOptions<any, any, any, any, string>' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & MutationExtraOptions<any, any, any, any, string>'.
                                                Type 'EndpointDefinitionWithQueryFn<any, any, any> & { [resultType]?: any; [baseQuery]?: any; } & { extraOptions?: unknown; } & QueryExtraOptions<any, any, any, any, string>' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                  Type 'Definitions[string]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                    Type 'Definitions[keyof Definitions]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                      Type 'Definitions[Definitions[keyof Definitions] extends MutationDefinition<any, any, any, any, string> ? keyof Definitions : never]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                        Type 'Definitions[MutationKeys<Definitions>]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                          Type 'Definitions[EndpointName]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                            Type 'Definitions[MutationKeys<Definitions>]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                              Type 'Definitions[Definitions[keyof Definitions] extends MutationDefinition<any, any, any, any, string> ? keyof Definitions : never]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
                                                                Type 'Definitions[keyof Definitions]' is not assignable to type 'EndpointDefinitionWithQuery<any, any, any>'.
```

[![](https://img.shields.io/badge/Formatted%20by-https%3A%2F%2Fst.elmah.io-%230da58e)](https://st.elmah.io)


### envirionment
TypeScirpt v4.5.4
node: 16.9
github actions container

# Reproduction
https://github.com/laststance/react-re-render-rules/runs/4494596354?check_suite_focus=true

--- a/.yarn/patches/console-testing-library__npm_0.3.1.patch
+++ b/.yarn/patches/console-testing-library__npm_0.3.1.patch
@@ -1,0 +1,26 @@
+diff --git a/index.d.ts b/index.d.ts
+index 1dbf70a6fdf4369188ad44166aee188c8efd2054..56b85e26b2b3030bc3407aab2b358fd4e5e39c6a 100644
+--- a/index.d.ts
++++ b/index.d.ts
+@@ -1,4 +1,4 @@
+-export const originalConsole = Console;
++export declare const originalConsole: Console;
+ 
+ type Options = {
+   isSilent?: boolean;
+@@ -24,14 +24,13 @@ export type TestingConsoleInstance = {
+   };
+   getRecord: (method: string) => string;
+   silence: boolean;
+-  private _targetConsole: TestingConsole | Console;
+ };
+ 
+ export function createConsole(options?: Options): TestingConsole;
+ 
+ export function mockConsole(
+   testingConsole: TestingConsole,
+-  targetConsoleParent?: {} = global,
++  targetConsoleParent?: typeof globalThis,
+   targetConsoleKey?: string
+ ): () => void;
+ 

--- a/package.json
+++ b/package.json
@@ -36,6 +36,7 @@
   },
   "resolutions": {
     "react-scripts": "patch:react-scripts@npm:4.0.2#.yarn/patches/react-scripts__npm_4.0.2.patch",
+    "console-testing-library": "patch:console-testing-library@npm:0.3.1#.yarn/patches/console-testing-library__npm_0.3.1.patch",
     "react-redux": "npm:7.2.4",
     "react": "npm:17.0.2",
     "react-dom": "npm:17.0.2"

--- a/packages/toolkit/src/query/core/module.ts
+++ b/packages/toolkit/src/query/core/module.ts
@@ -148,13 +148,21 @@ declare module '../apiTypes' {
         getRunningOperationPromise<EndpointName extends QueryKeys<Definitions>>(
           endpointName: EndpointName,
           args: QueryArgFrom<Definitions[EndpointName]>
-        ): QueryActionCreatorResult<Definitions[EndpointName]> | undefined
+        ):
+          | QueryActionCreatorResult<
+              Definitions[EndpointName] & { type: 'query' }
+            >
+          | undefined
         getRunningOperationPromise<
           EndpointName extends MutationKeys<Definitions>
         >(
           endpointName: EndpointName,
           fixedCacheKeyOrRequestId: string
-        ): MutationActionCreatorResult<Definitions[EndpointName]> | undefined
+        ):
+          | MutationActionCreatorResult<
+              Definitions[EndpointName] & { type: 'mutation' }
+            >
+          | undefined
 
         /**
          * A Redux thunk that can be used to manually trigger pre-fetching of data.

--- a/packages/toolkit/tsconfig.test.json
+++ b/packages/toolkit/tsconfig.test.json
@@ -11,7 +11,7 @@
     "target": "es2018",
     "jsx": "react",
     "baseUrl": ".",
-    "skipLibCheck": true,
+    "skipLibCheck": false,
     "noImplicitReturns": false
   }
 }

